R/summary.R

```diff
@@ -692,10 +692,10 @@ summary.gs_design <- function(object,
   # Save the full alpha as an attribute of the output summary table
   attr(output, "full_alpha") <-
     if (is.null(object$input$alpha)) {
-      if (is.null(object$upar$total_spend)) {
+      if (is.null(object$input$upar$total_spend)) {
         0.025
       } else {
-        object$upar$total_spend
+        object$input$upar$total_spend
```

Comment:
This line is causing the vignette `story-nph-futility.Rmd` to fail. `upar` is a numeric vector, not a named list

```R
wieand <- gs_power_ahr(
  enroll_rate = enroll_rate, fail_rate = fail_rate,
  upper = gs_b, upar = c(rep(Inf, 2), qnorm(.975)),
  lower = gs_b, lpar = c(0, 0, -Inf),
  event = 512 * c(.5, .75, 1)
)

str(wieand$input$upar)
##  num [1:3] Inf Inf 1.96
```

@LittleBeannie could you please investigate? Do we need to update to give `upar` names, or do we need to change how the total spend is extracted?

https://github.com/Merck/gsDesign2/blob/ee7806593e40e0887dec5d0b899500d69feff35c/vignettes/articles/story-nph-futility.Rmd#L89-L96

Reply:
@LittleBeannie The code below reproduces the error locally for me. The message matches the error in the failing pkgdown job. Could you please run it locally to see if you can also reproduce the error?

```R
remotes::install_github("Merck/gsDesign2@summary_upar", upgrade = FALSE)
library("gsDesign2")
enroll_rate <- define_enroll_rate(duration = 12, rate = 680 / 12)
fail_rate <- define_fail_rate(
  duration = c(3, 100),
  fail_rate = -log(.5) / 12,
  hr = c(1, .693),
  dropout_rate = 0
)
wieand <- gs_power_ahr(
  enroll_rate = enroll_rate, fail_rate = fail_rate,
  upper = gs_b, upar = c(rep(Inf, 2), qnorm(.975)),
  lower = gs_b, lpar = c(0, 0, -Inf),
  event = 512 * c(.5, .75, 1)
)
summary(wieand)
## Error in object$input$upar$total_spend :
##   $ operator is invalid for atomic vectors
```

Reply:
Thanks for the confirmation, @jdblischak . I tried it again and it runs without error.
![image](https://github.com/Merck/gsDesign2/assets/43153957/8277db91-d3e0-4937-af60-2d44a3b3333b)

Reply:
@LittleBeannie please run the following. Does your code have the `input` between `object` and `upar`?

```R
grep("total_spend", deparse(gsDesign2:::summary.gs_design), value = TRUE)
## [1] "        if (is.null(object$input$upar$total_spend)) {"
## [2] "            object$input$upar$total_spend"  
```

Reply:
Here is my output:
![image](https://github.com/Merck/gsDesign2/assets/43153957/0be5c6a2-2a7d-4dc2-a8a9-496fb34e4812)

Reply:
You aren't using the code in this PR. @yuliasidi updated the lines to be `object$input$upar$total_spend`. You won't observe the error unless you install the code from this PR

```R
remotes::install_github("Merck/gsDesign2@summary_upar", upgrade = FALSE)
```

Reply:
My bad... I installed Yulia's branch and run the above codes locally, and get the following error. I have get it fixed in my latest commit.
![image](https://github.com/Merck/gsDesign2/assets/43153957/6608468f-d1cf-4533-9782-d0ce93d9a660)